src/main/java/no/ndla/taxonomy/rest/v1/TopicResources.java

```diff
@@ -93,15 +92,17 @@ public ResponseEntity<Void> post(
             @Parameter(name = "connection", description = "new topic/resource connection ") @RequestBody AddResourceToTopicCommand command) {
 
         Node topic = nodeRepository.getByPublicId(command.topicid);
-        Resource resource = resourceRepository.getByPublicId(command.resourceId);
+        Node resource = nodeRepository.getByPublicId(command.resourceId);
         Relevance relevance = command.relevanceId != null ? relevanceRepository.getByPublicId(command.relevanceId)
                 : null;
 
-        final NodeResource topicResource;
-        topicResource = connectionService.connectNodeResource(topic, resource, relevance, command.primary,
-                command.rank == 0 ? null : command.rank);
+        var primary = Optional.of(command.primary);
+        var rank = command.rank == 0 ? null : command.rank;
 
-        URI location = URI.create("/topic-resources/" + topicResource.getPublicId());
+        final NodeConnection topicResource;
+        topicResource = connectionService.connectParentChild(topic, resource, relevance, rank, primary);
+
+        URI location = URI.create("/node-child/" + topicResource.getPublicId());
```

Comment:
Denne må vel også beholdes.
```suggestion
        URI location = URI.create("/topic-resources/" + topicResource.getPublicId());
```